components/director/internal/domain/scenarioassignment/service.go

```diff
@@ -126,12 +141,22 @@ func (s *service) List(ctx context.Context, pageSize int, cursor string) (*model
 	return s.repo.List(ctx, tnt, pageSize, cursor)
 }
 
-func (s *service) DeleteForSelector(ctx context.Context, selector model.LabelSelector) error {
+func (s *service) DeleteManyForSameSelector(ctx context.Context, in []*model.AutomaticScenarioAssignment) error {
```

Comment:
why on the resolver level we have to call our service twice?
```
	assignments, err := r.svc.ListForSelector(ctx, selector)
	if err != nil {
		return nil, errors.Wrapf(err, "while getting the Assignments for selector [%v]", selector)
	}

	err = r.svc.DeleteManyForSameSelector(ctx, assignments)
	if err != nil {
		return nil, errors.Wrapf(err, "while deleting the Assignments for selector [%v]", selector)
	}
```
Then it looks strange that we ask the same service for some data, and on the `Delete` method we validate again that data.

Reply:
Unfortunately the `ListForSelector` is needed to return all assignments as a result of the resolver.

I had two options:
- pass the listed assignments to the service again
- query database twice :/

I picked the second option. Do you have any other ideas?